app/data_model/answer_store.py

```diff
@@ -29,11 +41,12 @@ def __eq__(self, other):
         return self.answer_map == other.answer_map
 
     @staticmethod
-    def _build_map(answers):
+    def _build_map(answers: List[Union[Answer, Dict]]):
+        """ Builds the answer_store's data structure from a list of Answer objects"""
         answer_map = {}
 
         for answer in answers:
-            answer_map[answer['answer_id']] = answer
+            answer_map[answer['answer_id'], answer.get('list_item_id')] = Answer.from_dict(answer)
```

Comment:
```suggestion
		answer_map = {(answer['answer_id'], answer.get('list_item_id')]): Answer.from_dict(answer) for answer in answers}
```

Reply:
Dict comp would be more efficient?

Reply:
Not sure how much more efficient it is, but certainly cleaner.